extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/DevServicesConfig.java

```diff
@@ -110,7 +110,7 @@ public class DevServicesConfig {
     /**
      * The `JAVA_OPTS` passed to the keycloak JVM
      */
-    @ConfigItem
+    @ConfigItem(defaultValue = "-XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90")
```

Comment:
I would expect the default values of the container to allow Keycloak to run?

If we have a problem there, we should probably report it to the Keycloak team?

Reply:
These values allow keycloak to run, but it may consume to much memory.

I presume, that they use defaults, which are good for their intended use case (long running containers with many connections), but are not optimal for devmode use case (containers are being spin up often, but for short time and with a single local client)

Reply:
@gsmet FYI, please see #41426, feedback from the Keycloak team is that the new GC policy is now used in Keycloak 25, with the advice being to set the limits at the (Quarkus) KC devservice level but that is hard to do right without impacting some systems. These suggested default properties are the one which were used in KC 24 and they worked for all type of systems

Reply:
I wonder, instead of trying to deal with all of these advanced memory management properties by default, should we document that you may have to set these or that properties in memory constrained environments ?

Reply:
@sberyozkin what are the downsides of implementing this change?

As I see it, it makes life easier for some of our users and doesn't make things worse for anybody. Putting this into documentation means, that some of the users have to do it themselves and spend more time and effort. Am I missing something?

Reply:
@fedinskiy Sorry for the delay, I'm a little bit concerned that we'll put the properties and will miss out on whatever happens in Keycloak around the GC management going forward, which may for example benefit unconstrained systems. My understanding that the problems happens on the QE system so far.

But, it would make sense IMHO to see if trying to control the limits explicitly can work and what difference it can make. Can you please ping @mabartos for the extra guidance and try to find the HEAP limits for this system as suggested at https://www.keycloak.org/server/containers#_specifying_different_memory_settings ?
Have a look please in the next week or so

Reply:
I've created a new PR for this with all the explanations: https://github.com/quarkusio/quarkus/pull/43601

Reply:
Thanks @mabartos